### ISSUE TITLE
for each event, show a google calendar "add to gcal" link

### ISSUE THREAD

**Issue by gregsadetsky** [gregsadetsky/nycnoise#33]
for each event shown on the homepage, add a "add to gcal" link

the link should use the sorta-documented format of calendar.google.com links available [here](https://github.com/InteractionDesignFoundation/add-event-to-calendar-docs/blob/main/services/google.md)

**Comment by evanconway** [comment]
@gregsadetsky Is this dependent on the [homepage event listing](https://github.com/gregsadetsky/nycnoise/issues/30) being complete? Trying to understand which part of the website this link should exist in.

**Comment by gregsadetsky** [comment]
it's definitely related and not independent, you're right! once we are showing the list of events on the homepage -- we want to add, for each event, an "add to google calendar" link next to each of these events

so yes, #30 first and then google calendar

but... you could also "prepare" (for this issue) by writing some of the code that converts a Python datetime object into a google calendar link/url....? that code would be purely python (and/or maybe django template-based-only?), and you could write it now + and then when #30 is done, it could be tackled onto the code...? that's one way to go about it

**Comment by evanconway** [comment]
Made a branch [](https://github.com/gregsadetsky/nycnoise/tree/gcal-event-link) for this. It's a bit ugly but I have some thoughts I want to flesh out first. The example provided is more of a link to create a calendar event on your own google calendar. But this feels a bit weird to me. Google calendar events seem setup to be events created by the host and links are shared with users. I'm wondering if instead we should ask hosts to provide a g-calendar link on creating an event, and simply provide that to users.

**Comment by gregsadetsky** [comment]
> Made [a branch](https://github.com/gregsadetsky/nycnoise/tree/gcal-event-link) for this. It's a bit ugly but I have some thoughts I want to flesh out first. The example provided is more of a link to create a calendar event on your own google calendar. But this feels a bit weird to me. Google calendar events seem setup to be events created by the host and links are shared with users. I'm wondering if instead we should ask hosts to provide a g-calendar link on creating an event, and simply provide that to users.

a few things:
- great work on getting this started!
- it's completely reasonable to run `eventToGCalLink` on each event as you do in the view code.
  - if you're interested, this could also be developed into a "template tag" (more info on what that is [here](https://docs.djangoproject.com/en/4.2/ref/templates/builtins/)). custom template tags is a bit of a more advanced django thing, but it could be nice to be able to have `{{ event.date | googleCalendarUrl }}` and that would just output the url right in the page markup -- I think it'd be very clean to have that computation done in the template, and that way you could remove it from the view code as well (less code = better) :-)
- finally, yes, it's a great idea for event organizers to submit google calendar events -- at my prodding, Jessica already started doing that (check the very top of https://nyc-noise.com/ -- specifically, see right after "submit: " the link to this [sample event](https://calendar.google.com/calendar/u/0/r/eventedit?text=*list+acts+separated+by+commas*&dates=20230102T010000Z/20230102T040000Z&details=%5Blink%5D+//+%5Bage+policy%5D+//+%5Baccessibility+info%5D+//+%5Boptional:+xyz+presents%5D&location=*find+and+select+venue*&sf=true&output=xml&add=contact@nyc-noise.com)) -- however, in Jessica's experience, events are submitted in a very very wide variety of formats - sometimes well formatted, sometimes missing info, etc. -- her curation and work is mostly about selecting events and formatting them well / in a presentable way for people to "consume" on the site

so tldr:
- definitely continue work on the gcal link -- I actually think you're ready to open a PR, it looks quite great and usable!
- let's keep things as is for now re: submitting events. that's a whole topic that we can get to later. for now, the django admin works.
